### PR TITLE
Fix documentation build

### PR DESCRIPTION
**Pull request by fjarri**
- Use `doc_cfg` and `doc_auto_cfg` during documentation build, to make the "available on crate feature <...>" label actually appear. [`doc_auto_cfg`](https://doc.rust-lang.org/unstable-book/language-features/doc-auto-cfg.html) in particular figures out the placement of such labels automatically, so I could remove it from the `Zeroize` impl.
- Fix imports in `impl_zeroize.rs`
- `cargo fmt` run (also the `reorder_imported_names` option in `rustfmt.toml` seems to have been deprecated)
 
You can test it locally by running `env RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --all-features`.

Currently in docs.rs:
Screen Shot 2023-01-02 at 12 42 25

Built with this PR:
Screen Shot 2023-01-02 at 12 42 47

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -67,6 +67,7 @@
 #![deny(missing_docs)]
 #![deny(meta_variable_misuse)]
 #![no_std]
+#![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
```
Comment (tony-iqlusion) [inline src/lib.rs:70]: If you switch to `doc_auto_cfg`, you no longer need `doc_cfg`, and by omitting `doc_cfg` you can also guard against people adding manual annotations (if docs are tested in CI, that is)

```suggestion
#![cfg_attr(docsrs, feature(doc_auto_cfg))]
```

**Comment by gakonst** [comment]
image

Hi @fjarri, do you have plans to get this over the line? We seem to be hitting some nightly docs error on CI https://github.com/informalsystems/tendermint-rs/pull/1285#issuecomment-1464139023

**Comment by fjarri** [comment]
Unfortunately I do not have control over this repo.

**Comment by gakonst** [comment]
oops my bad - cc @fizyk20

**Comment by novacrazy** [comment]
Backported this to 1.14.7

**Pull request merged**